frontend/src/views/RoomSelectView.vue

```diff
@@ -21,20 +43,66 @@ import { useCurrentRoomStore } from "@/store";
 const roomData = ref<Room[]>([]);
 const router = useRouter();
 const currentRoomStore = useCurrentRoomStore();
+const isLoading = ref(false);
+const retryCount = ref(0);
+const maxRetries = 3;
 
 async function fetchRooms() {
+  if (isLoading.value) return;
+  
+  isLoading.value = true;
   try {
     const response = await apiClient.getRooms();
     if (response.success) {
       // ルームIDの昇順でソート
       roomData.value = response.data.sort((a: Room, b: Room) => a.roomId - b.roomId);
-      console.log(response.data);
+      console.log("Successfully fetched rooms:", response.data);
+      retryCount.value = 0; // 成功したらリトライカウントをリセット
     } else {
       console.error("Failed to fetch rooms:", response.data);
+      await handleFetchError(response.status);
     }
   } catch (error) {
     console.error("Error fetching rooms:", error);
+    await handleFetchError(0);
+  } finally {
+    isLoading.value = false;
+  }
+}
+
+async function handleFetchError(status: number) {
+  // 認証エラー（401）またはユーザーが見つからない（404）の場合はユーザー選択画面に戻る
+  if (status === 401 || status === 404) {
+    console.warn("Authentication failed, redirecting to signup");
+    alert("認証に失敗しました。ユーザー選択画面に戻ります。");
+    router.push("/");
+    return;
+  }
+
+  // その他のエラーでリトライ可能な場合は再試行
+  if (retryCount.value < maxRetries) {
+    retryCount.value++;
+    console.log(`Retrying to fetch rooms (attempt ${retryCount.value}/${maxRetries})`);
+    setTimeout(() => {
+      fetchRooms();
+    }, 1000 * retryCount.value); // 指数バックオフ的に待機時間を増加
```

Comment:
```suggestion
    }, 1000 * retryCount.value); // 線形バックオフ的に待機時間を増加
```
厳密に言えばこの実装は指数（1, 2, 4, 8, ...）ではなく線形（1, 2, 3, 4, ...）のバックオフみたいです